### PR TITLE
Improve IO code

### PR DESCRIPTION
**Pull request by Bond-009**
* Style changes
* Remove remnants of SMB support
* Use `GetInvalidFileNameChars` instead of rolling our own
* Remove possible unexpected behaviour with async file streams
* Remove some dead code

### CODE REVIEW

File: Emby.Server.Implementations/IO/ManagedFileSystem.cs [hunk 1]
```diff
@@ -92,20 +73,25 @@ public virtual string ResolveShortcut(string filename)
...
 
         public virtual string MakeAbsolutePath(string folderPath, string filePath)
         {
-            if (string.IsNullOrWhiteSpace(filePath)) return filePath;
+            if (string.IsNullOrWhiteSpace(filePath))
+            {
+                return filePath;
+            }
+
+            if (filePath.Contains("://"))
```
Comment (JustAMan) [inline Emby.Server.Implementations/IO/ManagedFileSystem.cs]: combine with previous `if`? like

```C#
            if (string.IsNullOrWhiteSpace(filePath) || filePath.contains("://"))
            {
                return filePath;
            }
```

File: Emby.Server.Implementations/IO/ManagedFileSystem.cs [hunk 2]
```diff
@@ -92,20 +73,25 @@ public virtual string ResolveShortcut(string filename)
...
+            {
+                return filePath; // stream
+            }
 
-            if (filePath.Contains(@"://")) return filePath; //stream
-            if (filePath.Length > 3 && filePath[1] == ':' && filePath[2] == '/') return filePath; //absolute local path
+            if (filePath.Length > 3 && filePath[1] == ':' && filePath[2] == '/')
```
Comment (JustAMan) [inline Emby.Server.Implementations/IO/ManagedFileSystem.cs]: this seems a Windows-only check, isn't there some standard mechanism in dotnet to check if given path is an absolute one?..

File: Emby.Server.Implementations/IO/ManagedFileSystem.cs [hunk 3]
```diff
@@ -45,20 +40,6 @@ public virtual void AddShortcutHandler(IShortcutHandler handler)
...
 
-        protected void SetInvalidFileNameChars(bool enableManagedInvalidFileNameChars)
-        {
-            if (enableManagedInvalidFileNameChars)
-            {
-                _invalidFileNameChars = Path.GetInvalidFileNameChars();
-            }
-            else
-            {
-                // Be consistent across platforms because the windows server will fail to query network shares that don't follow windows conventions
```
Comment (JustAMan) [inline Emby.Server.Implementations/IO/ManagedFileSystem.cs]: are you sure this is an invalid use case?..

File: Emby.Server.Implementations/IO/StreamHelper.cs [hunk 4]
```diff
@@ -109,64 +109,6 @@ public async Task<int> CopyToAsync(Stream source, Stream destination, Cancellati
             }
         }
 
-        public async Task<int> CopyToAsyncWithSyncRead(Stream source, Stream destination, CancellationToken cancellationToken)
```
Comment (JustAMan) [inline Emby.Server.Implementations/IO/StreamHelper.cs]: were these dead code?

File: Emby.Server.Implementations/IO/ThrottledStream.cs [hunk 5]
```diff
@@ -1,355 +0,0 @@
-using System;
```
Comment (JustAMan) [inline Emby.Server.Implementations/IO/ThrottledStream.cs]: why this file is gone?

File: Emby.Server.Implementations/IO/StreamHelper.cs (hunk 4 above)
Comment (Bond-009) [reply Emby.Server.Implementations/IO/StreamHelper.cs]: Yep

File: Emby.Server.Implementations/IO/ThrottledStream.cs (hunk 5 above)
Comment (Bond-009) [reply Emby.Server.Implementations/IO/ThrottledStream.cs]: 1) Wasn't being used
2) Shouldn't be used as it's a big hack

File: Emby.Server.Implementations/IO/ManagedFileSystem.cs (hunk 3 above)
Comment (Bond-009) [reply Emby.Server.Implementations/IO/ManagedFileSystem.cs]: `GetInvalidFileNameChars` should do the trick, if not it's a bug in the core lib ;).
This was probably for when we were still handling network shares ourselves.

File: Emby.Server.Implementations/IO/ManagedFileSystem.cs (hunk 2 above)
Comment (Bond-009) [reply Emby.Server.Implementations/IO/ManagedFileSystem.cs]: There is in the new version of .Net core, not in the one we are using.

**Pull request merged**